gensim/models/word2vec.py

```diff
@@ -1610,13 +1612,16 @@ def accuracy(self, questions, restrict_vocab=30000, most_similar=most_similar, u
                     logger.debug("skipping line #%i with OOV words: %s" % (line_no, line.strip()))
                     continue
 
+                original_vocab = self.vocab
+                self.vocab = ok_vocab
                 ignore = set(self.vocab[v].index for v in [a, b, c])  # indexes of words to ignore
                 predicted = None
                 # find the most likely prediction, ignoring OOV words and input words
                 sims = most_similar(self, positive=[b, c], negative=[a], topn=False, restrict_vocab=restrict_vocab)
+                self.vocab = original_vocab
                 for index in matutils.argsort(sims, reverse=True):
```

Comment:
The exact same functionality is present as part of `most_similar` in case `topn` is true valued. Would it be better to simply pass `topn` as 1 and do a string comparison? Can get rid of `ok_index` and `ignore` in that case too